geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/DestroyJndiBindingFunction.java

```diff
@@ -36,9 +36,12 @@ public CliFunctionResult executeFunction(FunctionContext<Object[]> context) {
     if (destroyingDataSource) {
       typeName = "Data source";
       DataSource dataSource = lookUpDataSource(jndiName);
```

Comment:
Another complication: when you call JNDIInvoker.getDataSource() it when return null for things that do exist (i.e. were created by create jndi-binding) but that can not be used as a "DataSource". It will also return null if it does not exist. In this context I think you need to know that the given name exists in the "dataSourceMap" but that it is not an instance of DataSource. If that is true then you can't destroy it but need to tell them to use destroy jndi-binding. No need to do the instanceof checks. You could add a method on JNDIInvoker like this:
public boolean existsInDataSourceMap(String name) {
  return dataSourceMap.containsKey(name);
}
You could then use this method to figure out if getting null back from getDataSource means it does not exist at all or if it exists but is not a "DataSource".